scanner/rev_cache.go

```diff
@@ -0,0 +1,30 @@
+package scanner
+
+import "sync"
+
+type RevCache struct {
+	m  map[string]string
+	mu *sync.Mutex
+}
+
+func NewRevCache(m map[string]string) *RevCache {
+	return &RevCache{
+		m:  m,
+		mu: &sync.Mutex{},
+	}
+}
+
+func (c *RevCache) GetName(sha string) (string, bool) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	rev, ok := c.m[sha]
+	return rev, ok
+}
+
+func (c *RevCache) Cache(sha, name string) {
```

Comment:
I find this a little confusing since sha->name would logically be a multi-map (shas can have multiple refs pointing at them).